dart/lib/flat_buffers.dart

```diff
@@ -30,8 +30,8 @@ class BufferContext {
 
   factory BufferContext.fromBytes(List<int> byteList) {
     Uint8List uint8List = _asUint8List(byteList);
-    ByteData buf = new ByteData.view(uint8List.buffer, uint8List.offsetInBytes);
-    return new BufferContext._(buf);
+    ByteData buf = ByteData.view(uint8List.buffer, uint8List.offsetInBytes);
+    return BufferContext._(buf);
```

Comment:
We should add an analysis_options.yaml file to cover this.

Reply:
Should I add the `pedantic` package as a dev dependency and use the latest `analysis_options.yaml` from this package?

Reply:
My personal preference would be to not do that, but it's probably fine either way. If the goal here is just to get rid of the `dev` tag from the environment constraint, we can probably skip this for now and clean up analysis rules in a separate PR. There's also some work to do to migrate this library to null-safety.

Reply:
Should we just copy the current [`analysis_options.1.9.0.yaml`](https://github.com/google/pedantic/blob/master/lib/analysis_options.1.9.0.yaml) or is a custom set of rules a better fit?

Reply:
If we're just eliminating `new`, we can just start with that lint (`unnecessary_new`). 

But I think we should probably just split this into a separate PR to avoid confusion. I strongly suspect there are other places the linter will pick up, because when I wrote this code `new` wasn't optional yet :)

Reply:
OK, sounds good :+1: 
If I find some time, I can also look into null safety migration.